compiler-cli/src/fs.rs

```diff
@@ -342,10 +342,34 @@ fn is_gleam_path(path: &Utf8Path, dir: impl AsRef<Utf8Path>) -> bool {
     )
 }
 
+fn is_gleam_build_dir(e: &ignore::DirEntry) -> Option<bool> {
```

Comment:
Let's just return a bool here and assume that there's no gleam.toml if we can't read it.